yapapi/services.py

```diff
@@ -113,16 +115,10 @@ class Service:
     Service specifications.
     """
 
-    def __init__(self, cluster: "Cluster", ctx: WorkContext, network_node: Optional[Node] = None):
-        """Initialize the service instance for a specific Cluster and a specific WorkContext.
-
-        :param cluster: a cluster to which this service instance of this service belongs
-        :param ctx: a work context object for executing commands on a provider that runs this
-            service instance.
-        """
-        self._cluster: "Cluster" = cluster
-        self._ctx: WorkContext = ctx
-        self._network_node: Optional[Node] = network_node
+    def __init__(self):
+        self._cluster: Optional["Cluster"] = None
```

Comment:
if they're not assigned values here, I'd move these to the class level...

Reply:
937b39c